CRM/Contribute/BAO/Contribution.php

```diff
@@ -2652,10 +2648,18 @@ protected static function repeatTransaction(&$contribution, $input, $contributio
       else {
         $contributionParams['financial_type_id'] = $templateContribution['financial_type_id'];
       }
-      foreach (['contact_id', 'currency', 'source'] as $fieldName) {
-        $contributionParams[$fieldName] = $templateContribution[$fieldName];
+      foreach (['contact_id', 'currency', 'source', 'amount_level', 'address_id'] as $fieldName) {
+        if (isset($templateContribution[$fieldName])){
+          $contributionParams[$fieldName] = $templateContribution[$fieldName];
+        }
+      }
+      if (!empty($recurringContribution['campaign_id'])) {
+        // CRM-17718 the campaign id on the contribution recur record should get precedence.
+        $contributionParams['campaign_id'] = $recurringContribution['campaign_id'];
+      }
+      if (!isset($contributionParams['campaign_id']) && isset($templateContribution['campaign_id'])) {
```

Comment:
@eileenmcnaughton this seems to be new in this PR? So we are saying that if the Recur object has a campaign we use that otherwise if there is a campaign in the template use that. I think that makes sense but might be worthwhile code documenting that this is a deliberate change

Reply:
@seamuslee001 so the recur part is just moved - I guess it didn't have to be but it seemed clearer. The fall back to templateContribution was not done except for A.net before... unless they passed in campaign_id as an input param

Reply:
Agree that the recur part is just moved its just this template override part that felt new

Reply:
OK - I've added a comment about falling back on the template value